radix-engine-tests/tests/db/substate_database_overlay.rs

```diff
@@ -442,54 +464,89 @@ fn create_database_contents_hash<D: SubstateDatabase + ListableSubstateDatabase>
     accumulator_hash
 }
 
-/// Runs the scenarios on an [`InMemorySubstateDatabase`] and a [`SingleSubstateDatabaseOverlay`] wrapping
+trait DatabaseComparisonScenarioCheck {
```

Comment:
Minor: Is it worth it using this trait as opposed to the ability to use closures?

Reply:
It is mostly just a way of avoiding needing to copy a massive 4 line fn closure definition across 4 lines.

This is only used for these two tests, tbh I’m tempted to rewrite them to duplicate this logic and it’d probably be clearer!

Thoughts welcome